File: stack/stack.c
#include "singly.h"

/* We will be using singly linked lists for stack, all of the functions can be found in the linked list folder */

/* All of the functions included in the singly.h file can be used here
 * append, pop, count, index, contains, clear, peekfirst, peeklast, remove, insert, delete, print */

void push(linkedList *, int);
int peek(linkedList *);
linkedList *createStack();
void printStack();

void push(linkedList *linkedListo, int data)
{
	append(linkedListo, data);
}

int peek(linkedList *linkedListo)
{
	return peekLast(linkedListo);
}

linkedList *createStack()
{
	linkedList *stack = createLinkedList();
	return stack;
}

void printStack(linkedList *linkedListo)
{
	printLinkedList(linkedListo);
}

int main()
{
	return 0;
}

File: stack/stack.py
from singly import linkedList
from singly import Node

# We will be using singly linked lists for stack, all of the methods can be found in the linked list folder */

# All of the methods included in the singly.py file can be used here
# append, pop, count, index, contains, clear, peekfirst, peeklast, remove, insert, delete, print

class stack(linkedList):
    def push(self, data):
        self.append(data)

    def peek(self):
        return self.peekLast()
